### PR TITLE
Update typed arrays documentation

### LINKED ISSUE #79066: Unable to create a typed array copy from an untyped array of valid elements
### Godot version

v4.0.3.stable.official [5222a99f5]

### System information

Windows 10

### Issue description

It seems impossible to create a typed array copy from an untyped array (with valid elements). There is one (and only one, at least that is exposed to gdscript) array constructor meant to creating typed arrays. This constructor takes as first parameter a base array for copying it's elements from. But it just doesn't work, or at least it doesn't work when the typed array type is custom script class_name (instead of a native class name). If this is the intended behavior, then the array documentation is lacking the mention that typed arrays can only be of native types. If this is not intended, then the constructor does not seem to be working as expected.

### Steps to reproduce

Create a script with the code bellow and run it. It will yield the same error in any of the attempts to use the typed array. constructor

```
extends Node
class MyClass: pass # inner class to create a type array of it
func _ready()->void:
	var a:Array = [MyClass.new()] # untyped array to copy into the typed one
	var b:Array[MyClass] # array typed with a class name

	# NONE of the methods bellow work. all of them will yield the runtime error "trying to assign an array of type "Array[MyClass]" to a variable of type "Array[MyClass]".
	# the stack trace is
#	E 0:00:03:0344   DELETEME.gd:12 @ _ready(): Attempted to assign an object of type 'RefCounted' into a TypedArray, which does not inherit from 'MyClass'.
#  <C++ Error>    Condition "!ClassDB::is_parent_class(object->get_class_name(), class_name)" is true. Returning: false
#  <C++ Source>   core/variant/container_type_validate.h:129 @ validate_object()
#  <Stack Trace>  DELETEME.gd:12 @ _ready()
	b = Array(a, TYPE_OBJECT, &'MyClass', null)
	b = Array(a, TYPE_OBJECT, &'RefCounted', null)
	b = Array(a, TYPE_OBJECT, &'Object', null)
	b = Array([]+a, TYPE_OBJECT, &'MyClass', null)
	b = Array(a.duplicate(), TYPE_OBJECT, &'MyClass', null)
	b = Array([]+a.duplicate(), TYPE_OBJECT, &'MyClass', null)
	b = Array(a.slice(0), TYPE_OBJECT, &'MyClass', null)
	b = Array(a.map(func(p):return p), TYPE_OBJECT, &'MyClass', null)
	b = Array(a.filter(func(p):return true), TYPE_OBJECT, &'MyClass', null)

	# this is the only method that works
	b = []
	for i in a: b.push_back(i)
```

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by dalexeev**
1. Extend the typed array constructor documentation.
2. Correct documentation of `get_typed_*()` methods.

The constructor is verbose and a bit confusing, I've seen questions about it on Discord several times.

* Closes #79066.

### CODE REVIEW

File: doc/classes/Array.xml [hunk 1]
```diff
@@ -317,19 +340,19 @@
...
 			<description>
-				Returns a class name of a typed [Array] of type [constant TYPE_OBJECT].
+				Returns the [b]native[/b] class name of the typed array if the built-in type is [constant TYPE_OBJECT]. Otherwise, this method returns an empty string.
 			</description>
 		</method>
 		<method name="get_typed_script" qualifiers="const">
 			<return type="Variant" />
 			<description>
-				Returns the script associated with a typed array tied to a class name.
+				Returns the script associated with the typed array. This method returns a [Script] instance or [code]null[/code].
```
Comment (dalexeev) [inline doc/classes/Array.xml:355]: Actually it returns `Object::null`, but I think it's a bug.

```gdscript
print(null) # <null>
print([].get_typed_script()) # <Object#null>
```

Comment (dalexeev) [reply doc/classes/Array.xml:355]: Now I'm not sure if this is a bug, this is the behavior of all `Ref<T>` properties and method return values. Should we draw the reader's attention to this difference?

```gdscript
var sprite := Sprite2D.new()
print(sprite.texture) # <Object#null>
print(sprite.texture == null) # true
print(is_same(sprite.texture, null)) # false
print(typeof(sprite.texture)) # 24
print(typeof(null)) # 0
```

**Comment by Mickeon** [comment]
Also see https://github.com/godotengine/godot/pull/69451 in the eventuality that it will be taken a look at.

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**